py/src/ai/h2o/sparkling/VersionComponents.py

```diff
@@ -20,13 +20,23 @@
 class VersionComponents(object):
 
     @staticmethod
-    def parseFromVersion(version):
+    def parseFromSparklingWaterVersion(version):
         match = re.search(r"^((\d+\.\d+).(\d+))(-(\d+))?-(\d+.\d+)$", version)
         result = VersionComponents()
         result.fullVersion = match.group(0)
         result.sparklingVersion = match.group(1)
         result.sparklingMajorVersion = match.group(2)
         result.sparklingMinorVersion = match.group(3)
         result.nightlyVersion = match.group(5)
-        result.sparkVersion = match.group(6)
+        result.sparkMajorVersion = match.group(6)
+        return result
+
+    @staticmethod
+    def parseFromPySparkVersion(version):
+        match = re.search(r"^(\d+.\d+).(\d+)(.([0-9A-Za-z]+))?$", version)
```

Comment:
nit: `.` will match to any character, `\.` will match just a dot. It's just a note, it won't have any effect on components parsing. The same nit you can find in the regexp above!

Reply:
Thanks, will fix!